flexx/app/_assetstore.py

```diff
@@ -433,16 +446,49 @@ def add_shared_data(self, name, data):
             raise TypeError('add_shared_data() name must be a str.')
         if name in self._data:
             raise ValueError('add_shared_data() got existing name %r.' % name)
-        if isinstance(data, str):
-            if data.startswith('file://'):
-                data = open(data.split('//', 1)[1], 'rb').read()
-            elif data.startswith(('http://', 'https://')):
-                data = urlopen(data, timeout=5.0).read()
-        if not isinstance(data, bytes):
-            raise TypeError('add_shared_data() data must be bytes.')
+        if isinstance(data, str) and data.startswith('file://'):
+            if not self.in_data_dir(data):
+                raise ValueError('add_shared_data() with a local file must exist '
+                                 'and be inside a data dir (see add_data_dir().')
+        elif isinstance(data, str) and data.startswith(('http://', 'https://')):
```

Comment:
I understand, and agree that we should inform our users accordingly, but being able to serve static files from the server is common enough that we should support it, I think. E.g. if I want to put some images in my (non-production) app, I would not want to have to read them in memory, or setup a tornado StaticFileHandler (let along nginx).

To summarize the current approach:
- users can specify "data dirs" using `app.assets.add_data_dir()`. Maybe this needs to be `flexx.config.data_dirs`?
- When a `file://....` is specified in `send_data`, `add_data` or `add_shared_data`, it is verified that the given file exists and is inside a data dir.
- When the file is requested (via `/flexx/static/...`), and handled by the StaticFileHandler, the real filename on the FS is looked up in the assetstore or session, and it is again verified that the file is inside a data dir.